core/src/main/scala/org/broadinstitute/dsde/rawls/entities/compact/CompactEntityProvider.scala

```diff
@@ -381,7 +381,39 @@ class CompactEntityProvider(requestArguments: EntityRequestArguments,
                             entityName: String,
                             newName: String,
                             parentContext: RawlsRequestContext
-  ): Future[Int] = ???
+  ): Future[Int] = {
+    // Check if the newName is the same as the old name
+    if (newName == entityName) {
+      throw new RawlsExceptionWithErrorReport(
+        errorReport = ErrorReport(StatusCodes.BadRequest, s"New name $newName is the same as the old name $entityName")
+      )
+    }
+    // Validate the newName
+    EntityUtils.validateEntityName(newName)
+    // Perform the rename in a transaction
+    val renameFuture = repository.dataSource.inTransaction { _ =>
+      for {
+        // Check if the newName already exists
+        newNameExists <- repository.queries.getEntity(workspaceId, entityType, newName)
+        _ = if (newNameExists.nonEmpty) {
+          throw new RawlsExceptionWithErrorReport(
+            errorReport = ErrorReport(StatusCodes.Conflict, s"$entityType/$newName already exists as an entity")
+          )
```

Comment:
nit: for compatibility, try to throw the same exceptions/error messages as thrown by LocalEntityProvider. Here, for instance, LocalEntityProvider throws `s"Destination ${entity.entityType} ${newName} already exists"`.

Where you _can_ be different in the code is that you can throw a `DataEntityException` instead of a `RawlsExceptionWithErrorReport`, which is a little terser/easier to read. When responding to the user, `DataEntityException`s are automatically converted to an ErrorReport at https://github.com/broadinstitute/rawls/blob/4b1c8fa2eec92095427433222caeaa0e1b07f534/core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/RawlsApiService.scala#L83.